akka-tools-persistence/src/main/scala/no/nextgentel/oss/akkatools/aggregate/GeneralAggregateBase.scala

```diff
@@ -48,6 +49,32 @@ abstract class GeneralAggregateBase[E:ClassTag, S <: AggregateStateBase[E, S]:Cl
   private val defaultErrorHandler = (errorMsg:String) => log.debug("No cmdFailed-handler executed")
 
 
+  protected override def onSnapshotOffer(offer : SnapshotOffer) : Unit = {
+    throw new Exception(s"Handling of snapshot recovery not defined, received ${offer.metadata}")

```

Comment:
Why throwing exception? Do it like this and it will just work out of the box:
`state = offer.snapshot.asInstanceOf[S]`